clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1122,13 +1141,26 @@ public long maybeUpdate(long now) {
 
             // Beware that the behavior of this method and the computation of timeouts for poll() are
             // highly dependent on the behavior of leastLoadedNode.
-            Node node = leastLoadedNode(now);
-            if (node == null) {
+            LeastLoadedNode leastLoadedNode = leastLoadedNode(now);
+
+            // Rebootstrap if needed and configured.
+            if (leastLoadedNode.node() == null
+                    && !leastLoadedNode.isAtLeastOneConnected()
```

Comment:
Could we add a method to `LeastLoadedNode` that checks `leastLoadedNode.node() == null && !leastLoadedNode.isAtLeastOneConnected()` since we use that here and in Admin client? It will make it more obvious in LeastLoadedNode that the boolean is only used when node is null. We could then remove `isAtLeastOneConnected` method perhaps.

Reply:
Great suggestion, done